<launch>

  <arg name="debug" default="false"/>
  <arg if="$(arg debug)" name="launch_prefix" value="xterm -rv -e gdb -ex run -args"/>
  <arg unless="$(arg debug)" name="launch_prefix" value=""/>

  <arg name="valgrind" default="false"/>
  <arg if="$(arg valgrind)" name="valgrind_launch_prefix" value="valgrind --leak-check=full"/>
  <arg unless="$(arg valgrind)" name="valgrind_launch_prefix" value=""/>

  <node pkg="blackboard" name="BlackBoard" type="blackboard" output="screen" launch-prefix="$(arg launch_prefix) $(arg valgrind_launch_prefix)">				
    <rosparam command="load" file="$(find blackboard)/config/blackboard.yaml" />
  </node>

</launch>
